packages/test/src/test-worker-lifecycle.ts

```diff
@@ -5,45 +5,98 @@
  * @module
  */
 import { Worker } from '@temporalio/worker';
+import { Runtime } from '@temporalio/worker';
 import test from 'ava';
 import { RUN_INTEGRATION_TESTS } from './helpers';
 import { defaultOptions, isolateFreeWorker } from './mock-native-worker';
 
 if (RUN_INTEGRATION_TESTS) {
   test.serial('Worker shuts down gracefully', async (t) => {
-    const worker = await Worker.create({
-      ...defaultOptions,
+    try {
+      const worker = await Worker.create({
+        ...defaultOptions,
+        shutdownGraceTime: '500ms',
+        taskQueue: 'shutdown-test',
+      });
+      t.is(worker.getState(), 'INITIALIZED');
+      const p = worker.run();
+      t.is(worker.getState(), 'RUNNING');
+      process.emit('SIGINT', 'SIGINT');
+      // Shutdown callback is enqueued as a microtask
+      await new Promise((resolve) => process.nextTick(resolve));
+      t.is(worker.getState(), 'STOPPING');
+      await p;
+      t.is(worker.getState(), 'STOPPED');
+      await t.throwsAsync(worker.run(), { message: 'Poller was already started' });
+    } finally {
+      // Make sure Runtime is reset after this test
+      if (Runtime._instance) await Runtime._instance.shutdown();
+      Runtime.instance();
```

Comment:
Why create a new instance in this test?
Shouldn't the runtime automatically be shutdown after this test?

Reply:
Actually, it is required only for the two mock tests. It automatically reset after native tests.

The thing is that the mock native worker doesn't register itself with Runtime, and obviously doesn't unregister either. Consequently, `Runtime.shutdownIfIdle()` doesn't gets called at the end of the mock tests, and therefore, the Runtime instance never get discarded.